### DISCUSSION TITLE
Region based object ban system

### DISCUSSION THREAD

**Discussion by moisterrific** [Pryaxis/TShock#2023] [Ideas]
I think it would be better if ItemBan, ProjBan, and TileBan are improved to be region specific. The current implementation is: **whatever that's banned is banned globally**. This works well for the most part but it's not the best way since server owners/admins may want to only ban specific items say in a PvP region to maintain balance, while allowing the items to be used everywhere else in PvE situations. My idea is to add regional sub command parameters on top of the existing way the aforementioned bans are handled. 

For example: if you do not specify a region, the item will be globally banned like before, but say you can do like `/itemban addregion zenith pvparena` to disallow users from using Zenith in a PvP region. Or `/itemban delregion <item> <region>` to remove a ban. This way tile breaking explosives can be denied from protected areas like TownNPC buildings while players can still use them to mine underground. 

I guess the global ban would take precedent over region ones, or perhaps add more commands like `/itemban allowregion <item> <region>` and vice versa. Or would that be too confusing?

Current implementation for reference:
- `/item|proj|tileban`
  - `add <item|proj ID|tile ID>`
  - `allow <item|proj ID|tile ID> <group>`
  - `del <item|proj ID|tile ID>`
  - `disallow <item|proj ID|tile ID> <group>`
  - `list [page]`

**Comment by Kojirremer** [comment]
I agree that it would be quite helpful if there was a vanilla TShock option for region banning of those. I'll mention, just in case, that at the very least item banning within a region can be currently done via the Region Trigger plugin.
